src/sagas/repository.ts

```diff
@@ -268,6 +268,21 @@ export function* getSnapshots({ payload }: any): any {
   }
 }
 
+export function* patchSnapshotDescription({ payload }: any): any {
```

Comment:
What does a `function*` do?

Reply:
This indicates the function is a Generator
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*

Generators are nice because they support yield
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/yield

This helps deal with managing callbacks.  There are other approaches, but this seems to be a good one for this codebase and is already the pattern in use.